### PR TITLE
[GOBBLIN-329]Add a new basic cluster integration test

### PR DESCRIPTION
**Pull request by HappyRay**
Run the hello world job and verify that the output files are generated.

It takes about 12 seconds to run locally by itself within IDEA and
4 seconds when running with others.

The test is modeled after the existing GobblinClusterKillTest test.

Also fixed a typo: GobblinClusterConfigurationKeys.CLUSTRER_WORK_DIR should be
GobblinClusterConfigurationKeys.CLUSTER_WORK_DIR

### CODE REVIEW

File: gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.cluster;
+
+
+import com.google.common.base.Optional;
```
Comment (htran1) [inline gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java]: Our import order convention is java, org, com, org.apache.gobblin.

File: gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    Config configFromResource = getConfigFromResource();
+    Config configOverride = getConfigOverride();
+    _config = configOverride.withFallback(configFromResource).resolve();
+  }
+
+  private Config getConfigOverride() {
+    Map<String, String> configMap = new HashMap<>();
+    String zkConnectionString = _testingZKServer.getConnectString();
+    configMap.put(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY, zkConnectionString);
+    configMap.put(GobblinClusterConfigurationKeys.CLUSTRER_WORK_DIR, _workPath.toString());
```
Comment (htran1) [inline gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java]: Existing typo. It is only referenced in one other file, so can you fix it?

File: gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java (hunk 1 above)
Comment (HappyRay) [reply gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java]: How is it enforced? 
I used intellij's import organization feature. It's inefficient to enforce this convention without automation.

Comment (htran1) [reply gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java]: There is an intellij code style template file at https://gobblin.readthedocs.io/en/latest/developer-guide/CodingStyle/. It needs to be updated for the org.apache.gobblin namespace.

File: gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java (hunk 2 above)
Comment (HappyRay) [reply gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java]: Fixed. Thanks

File: gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java (hunk 1 above)
Comment (HappyRay) [reply gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java]: Fixed.

**Pull request closed without merging**